src/renderer/pages/Onboarding/Vault/DdKeyQrReader/DdKeyQrReader.tsx

```diff
@@ -0,0 +1,284 @@
+import { useState } from 'react';
+
+import { QrReader } from '@renderer/components/common';
+import {
+  DdAddressInfoDecoded,
+  DdSeedInfo,
+  ErrorObject,
+  QrError,
+  VideoInput,
+} from '@renderer/components/common/QrCode/common/types';
+import { Icon, Loader, Button, CaptionText, FootnoteText, Select, SmallTitleText } from '@renderer/shared/ui';
+import { DropdownOption, DropdownResult } from '@renderer/shared/ui/Dropdowns/common/types';
+import { useI18n } from '@renderer/app/providers';
+import { cnTw } from '@renderer/shared/lib/utils';
+import { WhiteTextButtonStyle } from '@renderer/components/common/QrCode/common/constants';
+
+const enum CameraState {
+  ACTIVE,
+  LOADING,
+  SELECT,
+  UNKNOWN_ERROR,
+  INVALID_ERROR,
+  DECODE_ERROR,
+  DENY_ERROR,
+}
+
+const CameraAccessErrors = [CameraState.UNKNOWN_ERROR, CameraState.DENY_ERROR, CameraState.DECODE_ERROR];
+
+const RESULT_DELAY = 250;
+
+type Props = {
+  size?: number | [number, number];
+  className?: string;
+  onGoBack: () => void;
+  onResult: (payload: DdAddressInfoDecoded[]) => void;
+};
+
+export const DdKeyQrReader = ({ size = 300, className, onGoBack, onResult }: Props) => {
+  const { t } = useI18n();
+
+  const [cameraState, setCameraState] = useState<CameraState>(CameraState.LOADING);
+  const [activeCamera, setActiveCamera] = useState<DropdownResult<string>>();
+  const [availableCameras, setAvailableCameras] = useState<DropdownOption<string>[]>([]);
+
+  const [isScanComplete, setIsScanComplete] = useState(false);
+  const [{ decoded, total }, setProgress] = useState({ decoded: 0, total: 0 });
+
+  const isCameraPending = CameraState.LOADING === cameraState;
+  const isCameraOn = !isCameraPending && !CameraAccessErrors.includes(cameraState);
+
+  const isCameraError = [
+    CameraState.UNKNOWN_ERROR,
+    CameraState.INVALID_ERROR,
+    CameraState.DECODE_ERROR,
+    CameraState.DENY_ERROR,
+  ].includes(cameraState);
+
+  const onCameraList = (cameras: VideoInput[]) => {
+    const formattedCameras = cameras.map((camera, index) => ({
+      //eslint-disable-next-line i18next/no-literal-string
+      element: `${index + 1}. ${camera.label}`,
+      value: camera.id,
+      id: camera.id,
+    }));
+
+    setAvailableCameras(formattedCameras);
+
+    if (formattedCameras.length > 0) {
+      setActiveCamera(formattedCameras[0]);
+      setCameraState(CameraState.ACTIVE);
+    }
+  };
+
+  // FIXME: camera is blocked after 3 denies (that's intended browser reaction)
+  // Set attempts counter and show special notification
+  const onRetryCamera = () => {
+    setCameraState(CameraState.LOADING);
+  };
+
+  const resetCamera = () => {
+    setActiveCamera(undefined);
+    setProgress({ decoded: 0, total: 0 });
+  };
+
+  const onScanResult = (qrPayload: DdSeedInfo[]) => {
+    try {
+      const derivations: DdAddressInfoDecoded[] = [];
+      qrPayload.forEach((qr) => {
+        if (qr.multiSigner) {
+          encodeAddress(qr.multiSigner.public);
+        }
```

Comment:
i think better to add comment, to better understand why we do encodeAddress and don't use result